From 7c1daf4d308cce7661f36b9d69f6f7a49478a9d2 Mon Sep 17 00:00:00 2001
From: Tien Nguyen <tien.nguyen.uh@renesas.com>
Date: Thu, 21 May 2026 08:43:12 +0000
Subject: [PATCH] mmngrbuf: update following kernel 6.18

Update the platform driver remove callback to return void, matching
the current struct platform_driver API.

Upstream-Status: Pending

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>
---
 .../mmngrbuf-module/files/mmngrbuf/drv/mmngr_buf_drv.c      | 6 ++----
 .../files/mmngrbuf/include/mmngr_buf_private.h              | 2 +-
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/drv/mmngr_buf_drv.c b/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/drv/mmngr_buf_drv.c
index 58f9c0b..75b60eb 100644
--- a/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/drv/mmngr_buf_drv.c
+++ b/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/drv/mmngr_buf_drv.c
@@ -411,13 +411,11 @@ static int mm_probe(struct platform_device *pdev)
 	return 0;
 }
 
-static int mm_remove(struct platform_device *pdev)
+static void mm_remove(struct platform_device *pdev)
 {
 	misc_deregister(&misc);
 
 	kfree(mm_buf_drvdata);
-
-	return 0;
 }
 
 static const struct of_device_id mm_of_match[] = {
@@ -449,4 +447,4 @@ module_init(mm_init);
 module_exit(mm_exit);
 
 MODULE_LICENSE("Dual MIT/GPL");
-MODULE_IMPORT_NS(DMA_BUF);
+MODULE_IMPORT_NS("DMA_BUF");
diff --git a/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/include/mmngr_buf_private.h b/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/include/mmngr_buf_private.h
index ccc41d7..b78e1b8 100644
--- a/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/include/mmngr_buf_private.h
+++ b/mmngr_drv/mmngrbuf/mmngrbuf-module/files/mmngrbuf/include/mmngr_buf_private.h
@@ -102,7 +102,7 @@ static int dmabuf_mmap(struct dma_buf *buf, struct vm_area_struct *vma);
 static int dmabuf_vmap(struct dma_buf *buf, struct iosys_map *map);
 static void dmabuf_vunmap(struct dma_buf *buf, struct iosys_map *map);
 static int mm_probe(struct platform_device *pdev);
-static int mm_remove(struct platform_device *pdev);
+static void mm_remove(struct platform_device *pdev);
 static int mm_init(void);
 static void mm_exit(void);
 static int mm_ioc_export_start(int __user *arg, struct MM_BUF_PRIVATE *priv);
-- 
2.43.0

